src/shortcuts/redo.ts

```diff
@@ -1,20 +1,30 @@
-import { Dispatch } from 'react'
 import RedoIcon from '../components/RedoIcon'
 import { Shortcut } from '../@types'
 import { isRedoEnabled } from '../selectors/isRedoEnabled'
-
-interface RedoAction {
-  type: 'redoAction'
-}
+import { alert as alertAction } from '../action-creators'
+import { NAVIGATION_ACTIONS } from '../constants'
 
 const redoShortcut: Shortcut = {
   id: 'redo',
   label: 'Redo',
   description: 'Redo',
   svg: RedoIcon,
-  exec: (dispatch: Dispatch<RedoAction>, getState) => {
+  exec: (dispatch, getState) => {
     if (!isRedoEnabled(getState())) return
+
     dispatch({ type: 'redoAction' })
+
+    const { inversePatches } = getState()
+
+    // Checks the last action type from inverse patch history. Checks from the state after dispatching a redo action.
+    const lastActionType = inversePatches[inversePatches.length - 1]?.[0]?.actions[0]
+
+    if (!lastActionType) return
+
+    // Ignore navigation actions
+    if (NAVIGATION_ACTIONS[lastActionType]) return
```

Comment:
Navigation actions get bundled up and undone along with the last modifying action. We still want to dispatch an alert, we just need to be smarter about it to grab the right action type.

For example, if the last actions are `editThought`, `cursorUp`, it will undo both of them and we need to alert "Undo: Edit".